internal/provider/resource_gitlab_pages_domain_test.go

```diff
@@ -0,0 +1,124 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabPagesDomain_basic(t *testing.T) {
+	var pagesDomain gitlab.PagesDomain
+	rInt := acctest.RandInt()
+	testAccCheck(t)
+	project := testAccCreateProject(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabPagesDestroy,
+		Steps: []resource.TestStep{
+			// Create a pages domain with all options
+			{
+				Config: testAccGitlabPagesDomainCreate(rInt, project.PathWithNamespace),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabPagesDomainExists("gitlab_pages_domain.this", &pagesDomain),
+					resource.TestCheckResourceAttrSet("gitlab_pages_domain.this", "auto_ssl_enabled"),
```

Comment:
Given the subsequent `Verify import` test step both of these checks are already done, because:

1. `import` does a `Read()` and therefore the Pages domain actually has to exist
2. it would result in a non-empty plan in case the `auto_ssl_enabled` is not properly set ...

Reply:
What is your recommendation? I would prefer to have a "create" test case, beside the import.

Reply:
Maybe this helps: [Use ImportStateVerify TestSteps as a replacement for writing custom "CheckExists" and "CheckAttributes" check funcs.](https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008#:~:text=and%20data%20sources.-,Use%20ImportStateVerify%20TestSteps%20as%20a%20replacement%20for%20writing%20custom%20%22CheckExists%22%20and%20%22CheckAttributes%22%20check%20funcs.,-Undesirable%3A)

Also see the best practice here: https://github.com/bflad/tfproviderlint/tree/main/passes/AT002#at002